applications/FluidDynamicsApplication/custom_processes/compute_one_side_nodal_pressure_gradient_process.cpp

```diff
@@ -0,0 +1,148 @@
+//
+//  Main authors:    Mohammad R. Hashemi
+//
+
+// Project includes
+#include "includes/define.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+#include "fluid_dynamics_application_variables.h"
+#include "compute_one_side_nodal_pressure_gradient_process.h"
+
+
+namespace Kratos
+{
+
+/* Public functions *******************************************************/
+
+/// constructor
+ComputeOneSideNodalPressureGradientProcess::ComputeOneSideNodalPressureGradientProcess(
+        ModelPart& rModelPart)
+    : Process(), mrModelPart(rModelPart) {}
+
+/// Constructor with Kratos parameters.
+ComputeOneSideNodalPressureGradientProcess::ComputeOneSideNodalPressureGradientProcess(
+    ModelPart& rModelPart,
+    Parameters Parameters)
+    : ComputeOneSideNodalPressureGradientProcess(
+        rModelPart
+    ){}
+
+/// Constructor with Kratos model
+ComputeOneSideNodalPressureGradientProcess::ComputeOneSideNodalPressureGradientProcess(
+    Model& rModel,
+    Parameters Parameters)
+    : ComputeOneSideNodalPressureGradientProcess(
+        rModel.GetModelPart(Parameters["model_part_name"].GetString())
+    ){}
+
+void ComputeOneSideNodalPressureGradientProcess::Execute(){
+
+    KRATOS_TRY;
+
+    // Set to zero
+    block_for_each(mrModelPart.Nodes(), [&](Node<3>& rNode){
+        rNode.SetValue(NODAL_AREA, 0.0);
+        rNode.FastGetSolutionStepValue(PRESSURE_GRADIENT).clear();
+    });
+
+    // Current domain size
+    const unsigned int num_dim = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
+    const unsigned int num_nodes = num_dim + 1; //For tetrahedra and triangles
+
+    // Auxiliar containers
+    double detJ0 = 0.0;
+    Matrix DN_DX, J0, InvJ0;
+    Vector N(num_nodes), pressures(num_nodes), distances(num_nodes), grad(num_dim);
+
+    // First element iterator
+    const auto it_element_begin = mrModelPart.ElementsBegin();
+
+    // Iterate over the elements
+    #pragma omp parallel for firstprivate(DN_DX, N, J0, InvJ0, detJ0, pressures, distances, grad)
+    for(int i_elem=0; i_elem<static_cast<int>(mrModelPart.Elements().size()); ++i_elem) {
+
+        auto it_elem = it_element_begin + i_elem;
+        auto& r_geometry = it_elem->GetGeometry();
+
+        // Current geometry information
+        const unsigned number_of_nodes = r_geometry.PointsNumber();
+
+        for(std::size_t i_node=0; i_node<number_of_nodes; ++i_node){
+            distances(i_node) = r_geometry[i_node].FastGetSolutionStepValue(DISTANCE);
+        }
+
+        unsigned int nneg=0, npos=0;
+        for(unsigned int i = 0; i < number_of_nodes; ++i)
+        {
+            if(distances(i) > 0) npos += 1;
+            else /* if(distances(i) < 0) */ nneg += 1;
+        }
+
+        if(nneg == 0 || npos == 0)
+        {
+            if (N.size() != number_of_nodes){
+                N.resize(number_of_nodes);
+                pressures.resize(number_of_nodes);
+                distances.resize(number_of_nodes);
+            }
```

Comment:
I think that these are not needed (the sizes are set when allocating the arrays).